tests/analyser/analyserexternalvariable.cpp

```diff
@@ -229,3 +229,22 @@ TEST(AnalyserExternalVariable, dependencyByName)
     EXPECT_EQ(dependency, externalVariable->dependency(model, "membrane", "Cm"));
     EXPECT_EQ(nullptr, externalVariable->dependency(model, "membrane", "X"));
 }
+
+TEST(AnalyserExternalVariable, coverage)
+{
+    auto externalVariable = libcellml::AnalyserExternalVariable::create(nullptr);
+
+    EXPECT_EQ(nullptr, externalVariable->variable());
+    EXPECT_EQ(false, externalVariable->addDependency(nullptr));
+
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+    auto other_model = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+
+    externalVariable = libcellml::AnalyserExternalVariable::create(model->component("membrane")->variable("V"));
+
+    externalVariable->addDependency(model->component("membrane")->variable("Cm"));
+
+    EXPECT_EQ(false, externalVariable->removeDependency(nullptr, "membrane", "Cm"));
+    EXPECT_EQ(false, externalVariable->removeDependency(model, "not_membrane", "Cm"));
+}
```

Comment:
If this is solely for coverage purposes and not testing an aspect of the API it should be placed in the coverage test file.

Reply:
I have seen that, but I am not sure I like it. When I work on something (e.g., Analyser or Generator), I like the idea of having all the tests related to it in the same place rather than scattered around.

Reply:
Just to note that I prefer to keep these tests added purely for coverage to the coverage test file. Makes it nice and clear what they are and why they exist compared to the normal tests which are often the best documentation around for how to use libCellML.

Reply:
I really don't like it. When I work on, say, the analyser, I run the analyser tests which included the coverage tests for the analyser. Now, I will need to run both the analyser tests and the coverage tests, not to mention that the coverage tests will run other coverage-related tests that are not relevant to my work. So, yeah, it doesn't make sense to me to scatter things around like that.

But 2-1, so here goes: [#b1d72b8](https://github.com/cellml/libcellml/pull/1080/commits/b1d72b8c9ef0838e304b42d2fee2e1bfeb4c263d). 😕

Now, surely, we want to be consistent, so we will have to do the same for some annotator, importer, and printer tests.